ngraph/core/reference/include/ngraph/runtime/reference/sum.hpp

```diff
@@ -41,34 +42,32 @@ namespace ngraph
             }
 
             template <typename T>
-            void sum(const T* arg,
-                     T* out,
-                     const Shape& in_shape,
-                     const AxisSet& reduction_axes,
-                     bool keep_dims)
+            void sum(const T* arg, T* out, const Shape& in_shape, const AxisSet& reduction_axes)
             {
-                auto out_shape = reduce(in_shape, reduction_axes, keep_dims);
-                CoordinateTransform output_transform(out_shape);
-                std::vector<T> cs(shape_size(out_shape));
+                auto out_shape = reduce(in_shape, reduction_axes, false);
```

Comment:
As I see in `log_softmax` the attribute `keep_dims` was `true`. At the current moment you set `false`. How does it work? Why don't we get any regressions?

Reply:
Good question. This bool only makes sense when you are calculating indexes using CoordinateTransform `index()`.
After applying reduction along axes, the shape of output will have reduced dimensions equal to 1. These dims do not really affect on how indexes of a tensor are calculated.

Let's see this with one example:

```
Shape in_shape{2, 2, 2}
reduct_axes = {1}

```
- Output shape with `keep_dims = true`: `Shape out_shape{2, 1, 2}`
- Output shape with  `keep_dims = false`: `Shape out_shape{2, 2}`

In any case, number of output elements will be 4, and indexes of output will go from 0 to 3.